src/Analysis/Ast/Impl/Indexing/IndexParser.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Python.Core.IO;
+using Microsoft.Python.Parsing;
+
+namespace Microsoft.Python.Analysis.Indexing {
+    internal sealed class IndexParser : IIndexParser {
+        private readonly ISymbolIndex _symbolIndex;
+        private readonly IFileSystem _fileSystem;
+        private readonly PythonLanguageVersion _version;
+        private readonly CancellationTokenSource _allProcessingCts = new CancellationTokenSource();
+
+        public IndexParser(ISymbolIndex symbolIndex, IFileSystem fileSystem, PythonLanguageVersion version) {
+            _symbolIndex = symbolIndex ?? throw new ArgumentNullException(nameof(symbolIndex));
+            _fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+            _version = version;
+        }
+
+        public void Dispose() {
+            _allProcessingCts.Cancel();
+            _allProcessingCts.Dispose();
+        }
+
+        public Task ParseAsync(Uri uri, CancellationToken parseCancellationToken = default) {
```

Comment:
`uri` is never used as an URI - it is always used as a string path representation. Calling code also has a string path and creates URI specifically for this method. I think we can simply pass a string, or maybe even provide a stream rather than open it in the method.

Reply:
Both IndexManager and SymbolIndex have dictionaries from Uri to something. So using Uri avoids having to think how to compare paths of different formats.
Do we have another standard way?

Reply:
@jakebailey Any reason why you used Uri for SymbolIndex?

Reply:
We have `PathUtils.NormalizePath` to get normalized file and directory paths. `Uri` is very general and expensive type. Also, I'm not sure if `Uri.Equals` takes care of system being case-sensitive.

Reply:
I used `Uri`s in my original indexer because at every surface, it was `Uri` that was given. The old LS parse event was a Uri/Ast pair, and when you query for document symbols, LSP gives a Uri. There wasn't a cost then, because in every case, the Uri was constructed before I got my hands on it.

Whether or not it should really be strings, I'm not sure. Since it's no longer the parse event that's producing an Ast, but something new here that scans / gets updates from the document buffer, maybe a string does make more sense. However, I worry about cases where the editor passes a Uri that differs from the string by case, then when that path is looked up in the dictionary, it no longer matches (as a dictionary lookup with a string key is going to be case sensitive).

Reply:
Doing a quick check shows that `Uri`s are case senstive. https://glot.io/snippets/f906uaojg0

Reply:
Comparer should be chosen based on the OS. Windows can have individual case-sensitive folders, but I don't think we should care.